src/sql/workbench/api/common/sqlExtHost.protocol.ts

```diff
@@ -850,17 +850,17 @@ export interface ExtHostNotebookShape {
 	$save(managerHandle: number, notebookUri: UriComponents, notebook: azdata.nb.INotebookContents): Thenable<azdata.nb.INotebookContents>;
 
 	// Session Manager APIs
-	$refreshSpecs(managerHandle: number): Thenable<azdata.nb.IAllKernels>;
+	$refreshSpecs(managerHandle: number): Thenable<azdata.nb.IAllKernels | undefined>;
 	$startNewSession(managerHandle: number, options: azdata.nb.ISessionOptions): Thenable<INotebookSessionDetails>;
-	$shutdownSession(managerHandle: number, sessionId: string): Thenable<void>;
+	$shutdownSession(managerHandle: number, sessionId: string): Promise<void>;
 
 	// Session APIs
 	$changeKernel(sessionId: number, kernelInfo: azdata.nb.IKernelSpec): Thenable<INotebookKernelDetails>;
 	$configureKernel(sessionId: number, kernelInfo: azdata.nb.IKernelSpec): Thenable<void>;
 	$configureConnection(sessionId: number, connection: azdata.IConnectionProfile): Thenable<void>;
 
 	// Kernel APIs
-	$getKernelReadyStatus(kernelId: number): Thenable<azdata.nb.IInfoReply>;
+	$getKernelReadyStatus(kernelId: number): Thenable<azdata.nb.IInfoReply | null>;
```

Comment:
why not using undefined here?

Reply:
we were already using null here, so I just undated the typings without a functional change.